script.js

```diff
@@ -27,127 +27,368 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
 
 my_widget_script =
-{
-  init:function (mode, json_data) {
-    //this method is called when the form is being constructed
-    // parameters
-    // mode = if it equals 'view' than it should not be editable
-    //        if it equals 'edit' then it will be used for entry
-    //        if it equals 'view_dev' same as view,  does some additional checks that may slow things down in production
-    //        if it equals 'edit_dev' same as edit,   does some additional checks that may slow things down in production
- 
-    // json_data will contain the data to populate the form with, it will be in the form of the data
-    // returned from a call to to_json or empty if this is a new form.
-    //By default it calls the parent_class's init.
- 
-    //TO DO write code specific to your form
-    this.parent_class.init(mode, json_data);
-    
-    if (mode.indexOf('view') > -1) {
-      var isEmpty = function(tr) {
-        var inputs = $('input', tr);
-        for (var i = 0; i < inputs.length; i++) {
-          if ($(inputs[i]).val()) {
-            return false;
+    {
+      init:function (mode, json_data) {
+        //this method is called when the form is being constructed
+        // parameters
+        // mode = if it equals 'view' than it should not be editable
+        //        if it equals 'edit' then it will be used for entry
+        //        if it equals 'view_dev' same as view,  does some additional checks that may slow things down in production
+        //        if it equals 'edit_dev' same as edit,   does some additional checks that may slow things down in production
+
+        // json_data will contain the data to populate the form with, it will be in the form of the data
+        // returned from a call to to_json or empty if this is a new form.
+        //By default it calls the parent_class's init.
+
+        //TO DO write code specific to your form
+        console.log('init function call starts')
+
+        // Number of attributes in each record
+        var nAttributes = 8;
+
+        moledata = '';
+        fwdata = '';
+
+        var json_obj = eval(json_data);
```

Comment:
Can't you just check the fields are populated after `this.parent_class.init(mode, json_data);`?? I don't see why you need to work with the JSON at all.